### PR TITLE
Update to 6.6

### PR DESCRIPTION
**Pull request by MrCry0**
Tested building with:
- qemuarm
- qemuarm64
- intel-corei7-64 (fixed)
- stm32* (fixed)

Some further cosmetic fixes will come after this PR is landed.

### CODE REVIEW

**Comment by quaresmajose** [comment]
https://github.com/foundriesio/lmp-kernel-cache/pull/163/commits/3527f858219bb28b2c44f7cf2135ba482b365364 `global: Replace "is not set" with =n
` have references to https://git.kernel.org/pub/scm/linux/kernel/git/masahiroy/linux-kbuild.git/commit/?h=for-next&id=55de8686df7ed2b5237867b130e30c728bbd9db4 but afaik this one is not merged yet upstream so we also need to somehow backport this patch into our kernel or not?

**Comment by MrCry0** [comment]
> [3527f85](https://github.com/foundriesio/lmp-kernel-cache/commit/3527f858219bb28b2c44f7cf2135ba482b365364) `global: Replace "is not set" with =n ` have references to https://git.kernel.org/pub/scm/linux/kernel/git/masahiroy/linux-kbuild.git/commit/?h=for-next&id=55de8686df7ed2b5237867b130e30c728bbd9db4 but afaik this one is not merged yet upstream so we also need to somehow backport this patch into our kernel or not?

@quaresmajose As it is mentioned in the still not merged commit by Masahiro, since commit `c955ccafc38e ("kconfig: fix .config dependencies")` the .config file is not used anymore for building, it translates to `include/config/auto.conf`. As you can see in the https://github.com/foundriesio/linux/blob/linux-6.6.y/scripts/kconfig/confdata.c#L701 all `=n` are converted to "is not set" and immediately drop from auto.conf file.

So that for now there are no functional changes between both =n and "not set" forms.

The mainline yocto-kernel-cache already converted all options to =n form, so I'd prefer to have our version of repo to be aligned with the mainline repo.

File: ktypes/preempt-rt/preempt-rt.cfg [hunk 1]
```diff
@@ -16,3 +131,950 @@ CONFIG_NET_SCH_ETF=y
...
+#
+# Hardware crypto devices
+#
+
+#
+# Library routines
+#
+CONFIG_LIBCRC32C=y
+CONFIG_ZLIB_DEFLATE=y
+>>>>>>> 32d6b478 (config: replace CONFIG_AUTOFS4_FS with CONFIG_AUTOFS_FS)
```
Comment (ricardosalveti) [inline ktypes/preempt-rt/preempt-rt.cfg:1080]: There is a conflict here, and it also adds a bunch of non related changes.

File: features/optee/optee.cfg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# SPDX-License-Identifier: MIT
```
Comment (ricardosalveti) [inline features/optee/optee.cfg:1]: We do have features/tee/tee.cfg, maybe we can just replace with this one?

File: bsp/genericarm64/genericarm64-standard.scc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# SPDX-License-Identifier: MIT
```
Comment (ricardosalveti) [inline bsp/genericarm64/genericarm64-standard.scc:1]: We do have:
```
generic-arm64/generic-arm64.scc
generic-arm64/generic-arm64.cfg
generic-arm64/generic-arm64-standard.scc
```

Let's drop this one from this list, and have it as part of the other clean-up that will follow.

File: arch/riscv64/riscv64.cfg [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Doesn't yet work right on RISC-V
-# CONFIG_DYNAMIC_FTRACE is not set
+CONFIG_DYNAMIC_FTRACE=n
```
Comment (ricardosalveti) [inline arch/riscv64/riscv64.cfg:2]: While it makes sense, I would prefer this change in particular to be separated in another PR, just so we can perform just this small change in meta-lmp and validate if the kernel config is indeed the same for all targets.

File: features/optee/optee.cfg (hunk 2 above)
Comment (MrCry0) [reply features/optee/optee.cfg:1]: Yes, I'm planning to switch to the mainline version with our changes later.
Are we going to keep this PR as simple as possible?

Comment (MrCry0) [reply features/optee/optee.cfg:1]: I dropped it in this PR.

File: ktypes/preempt-rt/preempt-rt.cfg (hunk 1 above)
Comment (MrCry0) [reply ktypes/preempt-rt/preempt-rt.cfg:1080]: Done

File: bsp/genericarm64/genericarm64-standard.scc (hunk 3 above)
Comment (MrCry0) [reply bsp/genericarm64/genericarm64-standard.scc:1]: Done

File: arch/riscv64/riscv64.cfg (hunk 4 above)
Comment (MrCry0) [reply arch/riscv64/riscv64.cfg]: Done

**Review by ricardosalveti — APPROVED** [review]
LGTM, sorry for the delay.
